### PR TITLE
Add classification for degree_type field

### PR DESCRIPTION
**Pull request by rikirenz**
Adds a classification for `degree_type` field in according to the schema: https://github.com/inspirehep/inspire-schemas/blob/master/inspire_schemas/records/elements/degree_type.json

### CODE REVIEW

**Comment by rikirenz** [comment]
I would like to know how we should match the data from the legacy system with the new data defined in the `degree_type.json` (https://github.com/inspirehep/inspire-schemas/blob/master/inspire_schemas/records/elements/degree_type.json).

This is the list with values and number of occurrences on the legacy system:
````
 2806 PhD
   2319 PHD
    103 Master
     90 MAS
     28 MASTER
     27 master
     18 Bachelor
     12 Phd
     11 UG
      8 Habilitation
      5 Thesis
      2 Diploma
      1 MAs
      1 Mas
      1 Laurea
```` 
and this is the list of the new values in the schema:
````
        other
        diploma
        bachelor
        laurea
        master
        phd
        habilitation
````
and this is the actual mapping that we are doing:
````
DEGREE_TYPES_MAP = {                                                         
         'Bachelor': 'Bachelor',                                                  
         'UG': 'Bachelor',                                                        
         'MAS': 'Master',                                                         
         'master': 'Master',                                                      
         'Master': 'Master',                                                      
         'PhD': 'PhD',                                                            
         'PHD': 'PhD',                                                            
}
````

this is the mapping that I would like to implement:

````
  PhD -> phd
  PHD -> phd
  Phd -> phd
  Master -> master
  MAS -> master
  MASTER -> master
  master -> master
  Bachelor -> bachelor
  UG -> bachelor
  Habilitation -> habilitation
  Thesis -> laurea
  Diploma -> diploma
  MAs -> master
  Mas -> master
  Laurea -> laurea
````
am I right? @kaplun  @michamos

**Comment by michamos** [comment]
almost, thesis -> other, not everybody is Italian :)

**Comment by kaplun** [comment]
You can actually simplify your mapping by simply perform the lower case transformation first. (Invenio is case-insensitive).

```
  phd -> phd
  bachelor -> bachelor
  ug -> bachelor
  habilitation -> habilitation
  thesis -> thesis
  diploma -> diploma
  mas -> master
  laurea -> thesis
```

**Comment by rikirenz** [comment]
Thanks! Do you think is necessary keep this field: https://github.com/inspirehep/inspire-next/pull/1894/files#diff-338f51def51c1adf539431d8baf989f6L466
IMO no, because it is not defined in the schema. But please confirm it.

**Comment by kaplun** [comment]
IMHO nope because with the above mapping you exhaust the above mentioned list of occurrences on legacy.

**Comment by rikirenz** [comment]
I was trying to understand how we convert 701 from schema to marc and I found this: https://github.com/inspirehep/inspire-next/blob/master/inspirehep/dojson/hepnames/fields/bd1xx.py#L483
IMO `_degree_type` is not part of the schema, it should be removed and replaced with `degree_type`. Could you give me some explanation about it? @michamos @kaplun

**Comment by kaplun** [comment]
Indeed you should look at `degree_type`. Originally `_degree_type` was intended to contain those values that didn't belong to the `enum`. But there shouldn't be any anymore.

File: tests/unit/dojson/test_dojson_hepnames.py [hunk 1]
```diff
@@ -623,13 +628,39 @@ def test_other_names_from_400__triple_a():
...
+                'g': 'phd',
+                'x': 991627,
+                'y': '1'
+            }
+        ]
+    }
+
+    result = hepnames2marc.do(snippet)
+
+    assert expected == result
```
Comment (rikirenz) [inline tests/unit/dojson/test_dojson_hepnames.py]: I changed the test in this way for 2 reasons:

1. It is more readable
2. In order to test all the 701 fields

Comment (jacquerie) [inline tests/unit/dojson/test_dojson_hepnames.py]: Ok, but you're based on an old branch: that test is already deleted in master.

Comment (rikirenz) [inline tests/unit/dojson/test_dojson_hepnames.py]: :hankey:

**Comment by annetteholtkamp** [comment]
Why do you want to get rid of laurea? It's more specific than thesis I think.

- Annette


On Jan 30, 2017, at 5:04 PM, Samuele Kaplun <notifications@github.com<mailto:notifications@github.com>> wrote:


You can actually simplify your mapping by simply perform the lower case transformation first. (Invenio is case-insensitive).

  phd -> phd
  bachelor -> bachelor
  ug -> bachelor
  habilitation -> habilitation
  thesis -> thesis
  diploma -> diploma
  mas -> master
  laurea -> thesis

**Comment by kaplun** [comment]
It just says that's Italian. But it doesn't say if it's master or bachelor (as much as thesis doesn't say it).

**Comment by michamos** [comment]
still, it's some data that we have, and that we decided to keep in the enum, so it should be mapped as `laurea -> laurea`.

**Comment by kaplun** [comment]
Right. Didn't remember we actually have it in the `enum`.

**Comment by kaplun** [comment]
As pointed out in https://github.com/inspirehep/inspire-schemas/pull/80 
`thesis` -> `other` (since it's not a degree type for real and doesn't bring real information).

**Comment by kaplun** [comment]
BTW, I am confused: @spirosdelviniotis have you took over this branch?

**Comment by spirosdelviniotis** [comment]
@kaplun I am working on top of master branch.
Should I work on top of this one?

**Comment by kaplun** [comment]
@spirosdelviniotis I guess not. Was just asking to fully understand.

**Pull request closed without merging**